src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -328,10 +331,20 @@ export class TerminalInstance implements ITerminalInstance {
 
 			return undefined;
 		});
-		this._instanceDisposables.push(dom.addDisposableListener(this._xterm.element, 'mouseup', (event: KeyboardEvent) => {
-			// Wait until mouseup has propagated through the DOM before
-			// evaluating the new selection state.
-			setTimeout(() => this._refreshSelectionContextKey(), 0);
+
+		this._instanceDisposables.push(dom.addDisposableListener(this._xterm.element, 'mousedown', (event: KeyboardEvent) => {
+			// Track down the mousedown event since we may not have the corresponding mouseup:
+			// in case the mouse is released outside of _xterm.element.
+			this._selectionStarted = true;
+		}));
+
+		// We need to listen to the mouseup event up to the document since the user may release the mouse button anywhere
+		// outside of _xterm.element.
+		this._instanceDisposables.push(dom.addDisposableListener(document, 'mouseup', (event: KeyboardEvent) => {
```

Comment:
I'd prefer if this listener was only active after a mousedown and disposed of itself on mouseup, otherwise this will run on every mouse click. You can probably do this all within the mousedown listener without the need to introduce more properties to `TerminalInstance`.

Also this listener will also make it very easy to implement https://github.com/Microsoft/vscode/issues/36271 :+1:

Reply:
Oh you're right: that's much better.